docs/Configuration.md

```diff
@@ -0,0 +1,106 @@
+
+To add and use the new secret:
+1. Add a new key value pair to `parameters.secrets.value` in [deploy.Bicep.ps1](../.azure/deployBicep.ps1), for example `myShinyNewPassword`:
+    ```powershell
+    # Add auto generated secrets to parameters
+    AddMemberPath $paramsJson "parameters.secrets.value" @{
+        dialogportenPgAdminPassword = (GeneratePassword -length 30).Password,
+        # Add the new password
+        myShinyNewPassword = (GeneratePassword -length 30).Password
+    }
+    ```
+    The new `myShinyNewPassword` will be available in [main.bicep](../.azure/main.bicep) through `secrets.myShinyNewPassword`. However, this password will be different every time the IaC runs which is why we don't want to use this value directly. The bicep module **MODULE NAME HERE** will add `myShinyNewPassword` to the specific environments key vault if it does not already exist.
+2. Use the secret by fetching it from the environment key vault `envKeyVault.getSecret('myShinyNewPassword')`.
+
+Removing or modifying autogenerated secrets is not supported through IaC.
+
+# 4. External secrets
+External secrets are secrets that require human intervention to configure and must be known ahead of time, before the IaC runs. This could be a 3rd party api key or certificates. 
+
+To be able to support both external secrets and an IaC that can spin up a brand new environment on the fly, we've introduced a convention based "source key vault". The source key vault should exclusively contain external secrets and must exist before the IaC runs as it depends on it. The source key vault could be used by multiple environments. 
+
+## 4.1. Naming convention
+When the IaC runs, it copies specific secrets from the source to the deployments target environment key vault based on the following naming convention:
+
+| **Convention type** | **Source key vault**                     | **Environment key vault** |
+|---------------------|------------------------------------------|---------------------------|
+| **Wildcard**        | dialogporten--*--[SecretKey]             | [SecretKey]               |
+| **Environment**     | dialogporten--[Environment]--[SecretKey] | [SecretKey]               |
+
+There are two conventions, one that targets a specific environment, and one wildcard that targets all environments that the source is connected to through IaC. The environment specific convention takes precedence over the wildcard convention. For example, say we are running an IaC that targets the `Stage` environment with the following keys in the source key vault:
+
+| **Source key vault**          | **Stage key vault** | **IsCopied** | **Description**                                         |
+|-------------------------------|---------------------|--------------|---------------------------------------------------------|
+| dialogporten--*--foo          |                     | false        | Environment specific "foo" convention takes precedence. |
```

Comment:
Er bare a-z, A-Z, 0-9 og bindestrek som kan brukes i keys. Foreslår `Any` i stedet for `*`

Reply:
Good catch